### PR TITLE
Kubernetes v1.22 integration for Enterprise Components

### PR DESCRIPTION
**Pull request by freecaykes**
## Description
Absorb, Elastic eck-operator version update and CRDs changes for eck-operator v1.7.1 and Prometheus operator to v0.40.0

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by hjiawei** [comment]
Please upload the eck-operator v1.7.1 container image to gcr unique-caldron-775 (and Tigera public registry before the release). @doublek help me with that last time.

**Comment by hjiawei** [comment]
Some nit comments in the private doc PR. You may want to sync the CRDs again if needed.

File: config/enterprise_versions.yml
Comment (freecaykes) [inline config/enterprise_versions.yml:100]: - [ ] note to move prometheus CRDs here as well

**Comment by hjiawei** [comment]
Not for this PR though. Do we want to upgrade the `kubectl` command line utility to a newer version so that we can catch the deprecation warnings/errors earlier?

https://github.com/tigera/operator/blob/master/Makefile#L300

File: config/enterprise_versions.yml
Comment (freecaykes) [reply config/enterprise_versions.yml:100]: https://github.com/tigera/operator/pull/1511/commits/14915a58f440a7e55058616da8a705c089a52b3f

**Review by hjiawei — APPROVED** [review]
LGTM

**Review by doublek — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
